### PR TITLE
X86 non deep sx power seq

### PR DESCRIPTION
**Pull request by vphirema**

### CODE REVIEW

File: subsys/power_seq/x86_non_dsx.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+#ifndef __X86_NON_DSX_H__
+#define __X86_NON_DSX_H__
+
+/* Delay in ms for pass through signals */
+#define POWER_EC_PCH_DSW_PWROK_DELAY_MS	100
+#define POWER_EC_PCH_RSMRST_DELAY_MS	10
+#define POWER_EC_PCH_SYS_PWROK_DELAY_MS	50
+#define POWER_EC_VR_EN_VCCIN_DELAY_MS	5
+#define POWER_EC_PCH_PM_PWRBTN_DELAY_MS	200
```
Comment (jackrosenthal) [inline subsys/power_seq/x86_non_dsx.h:15]: should these be kconfig options?

Comment (vphirema) [reply subsys/power_seq/x86_non_dsx.h:15]: thank you for the feedback Jack.
These are constants needed by PCH, they don't change.

**Comment by sjg20** [comment]
How do we call / configure this code? Is there an example?

File: subsys/power_seq/x86_non_dsx.c [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+ */
+
+/* On power-on start boot up sequence */
+static enum power_states_ndsx power_state = SYS_POWER_STATE_G3S5;
+
+/* Implement in board file */
+static void enable_power_rails(bool enable)
+{
+	if (enable) {
+		gpio_set_level(AP_5P0V_A_RAIL, 1);
```
Comment (jettr) [inline subsys/power_seq/x86_non_dsx.c:14]: We should probably define these gpios in a power-sequence device in the device tree. This code would use that device information and expect certain gpio names to be populated in the device tree, like en-ap-5p0v-a. Then this cod would look that up and use the gpio_* Zephyr API (e.g `gpio_pin_get_raw` `gpio_port_set_bits_raw` and `gpio_port_clear_bits_raw`) to interact with the gpios.

File: subsys/espi/ndsx_espi.c [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	case ESPI_VWIRE_SIGNAL_SLP_S3:
+	case ESPI_VWIRE_SIGNAL_SLP_S4:
+		k_thread_resume(&pwrseq_thread_id);
+		break;
+	default:
+		break;
+	}
+}
+
+static void espi_bus_handler(const struct device *dev,
```
Comment (jettr) [inline subsys/espi/ndsx_espi.c:43]: The code in this function seems generic to eSPI. I would expect this to be in the espi driver itself

File: subsys/espi/ndsx_espi.c [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	if (power_gpio_sig_get(POWER_PCH_EC_SLP_SUS_N)) {
+		LOG_INF("Toggle PM PWRBTN");
+
+		gpio_set_level(EC_PCH_PWRBTN_N, 0);
+		k_msleep(POWER_EC_PCH_PM_PWRBTN_DELAY_MS);
+		power_gpio_sig_set(EC_PCH_PWRBTN_N, 1);
+	}
+}
+
+static void espi_vw_handler(struct espi_event *event)
```
Comment (jettr) [inline subsys/espi/ndsx_espi.c:28]: The code in this function sees like it it target to the power sequencing part. I think this code should go in the power sequence folder (in the same file as above espi power sequencing stuff)

File: subsys/espi/ndsx_espi.c [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+#include <drivers/espi.h>
+
+#include "x86_non_dsx.h"
+
+static const struct device *espi_dev;
+static struct espi_callback espi_bus_cb;
+static struct espi_callback espi_chan_cb;
+static struct espi_callback espi_vw_cb;
+
+static void espi_bus_reset(void)
```
Comment (jettr) [inline subsys/espi/ndsx_espi.c:16]: The code in this function sees like it it target to the power sequencing part. I think this code should go in the power sequence folder (maybe as a separate file that has to do with eSPI though)

File: subsys/espi/ndsx_espi.c [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+	espi_add_callback(espi_dev, &espi_chan_cb);
+
+	espi_init_callback(&espi_vw_cb, espi_bus_handler,
+			ESPI_BUS_EVENT_VWIRE_RECEIVED);
+	espi_add_callback(espi_dev, &espi_vw_cb);
+}
+
+u8_t vw_get_level(enum espi_vwire_signal signal)
+{
+	u8_t level;
```
Comment (jettr) [inline subsys/espi/ndsx_espi.c:100]:
```suggestion
	uint8_t level;
```
FYI, Zephyr is trying to more to the C99 types, so unit8_t and the like

**Review by jettr — CHANGES_REQUESTED** [review]

File: subsys/power_seq/x86_non_dsx.c (hunk 2 above)
Comment (vphirema) [reply subsys/power_seq/x86_non_dsx.c]: > How do we call / configure this code? Is there an example?

Added sample code

Comment (vphirema) [reply subsys/power_seq/x86_non_dsx.c]: > We should probably define these gpios in a power-sequence device in the device tree. This code would use that device information and expect certain gpio names to be populated in the device tree, like en-ap-5p0v-a. Then this cod would look that up and use the gpio_* Zephyr API (e.g `gpio_pin_get_raw` `gpio_port_set_bits_raw` and `gpio_port_clear_bits_raw`) to interact with the gpios.

ACK, added device tree now, I will work on adding follow-on CLs. Try to learn this new interface

File: subsys/espi/ndsx_espi.c (hunk 3 above)
Comment (vphirema) [reply subsys/espi/ndsx_espi.c]: ACK, I will add in follow on CLs

File: subsys/espi/ndsx_espi.c (hunk 4 above)
Comment (vphirema) [reply subsys/espi/ndsx_espi.c]: ACK

File: subsys/espi/ndsx_espi.c (hunk 5 above)
Comment (vphirema) [reply subsys/espi/ndsx_espi.c]: ACK

File: subsys/espi/ndsx_espi.c (hunk 6 above)
Comment (vphirema) [reply subsys/espi/ndsx_espi.c]: ACK

File: include/power/ndsx_espi.h [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * @brief Configure eSPI channels for Non Deep Slwwp well platforms
+ */
+void ndsx_espi_configure(void);
+
+/**
+ * @brief Return Virtual wire value
+
+ * @retval Acatual Virtual Wire value or 0 on failure
```
Comment (jettr) [inline include/power/ndsx_espi.h:18]: s/Acatual/Actual/

Should we return an int and use a negative return to indicate an error instead?

File: include/power/x86_non_dsx.h [hunk 8]
```diff
@@ -104,6 +104,9 @@ enum power_states_ndsx {
 #define POWER_EC_PCH_RSMRST_DELAY_MS	10
 #define POWER_EC_PCH_SYS_PWROK_DELAY_MS	50
 #define POWER_EC_VR_EN_VCCIN_DELAY_MS	5
+#define POWER_EC_PCH_PM_PWRBTN_DELAY_MS	200
+
+void espi_bus_reset(void);
```
Comment (jettr) [inline include/power/x86_non_dsx.h:109]: The prefix for this file `non_dsx.h` doesn't match the other in this CL : `ndsx`

File: subsys/power/x86_non_dsx.c [hunk 9]
```diff
@@ -244,11 +239,24 @@ static void power_pass_thru_handler(const char *in_signal,
 	}
 }
 
+void espi_bus_reset(void)
+{
+	/* If SOC is up toggle the PM_PWRBTN pin */
+	if (gpio_get_level(GPIO_NET_NAME(PCH_EC_SLP_SUS_L))) {
+		LOG_INF("Toggle PM PWRBTN");
+
+		gpio_set_level(GPIO_NET_NAME(EC_PCH_PWR_BTN_ODL), 0);
```
Comment (jettr) [inline subsys/power/x86_non_dsx.c:248]: This is where we should be using device tree to reference the gpios that the power sequence node listed

File: include/dt-bindings/gpio/gpio.h [hunk 10]
```diff
@@ -7,6 +7,8 @@
 #ifndef ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_GPIO_H_
 #define ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_GPIO_H_
 
+#include <drivers/gpio_flags.h>
```
Comment (jettr) [inline include/dt-bindings/gpio/gpio.h:10]: So I have been wanting a similar change to Zephyr, but I think we need to get a more general concensus on making this change. The `gpio_dt_flags_t` is only 8-bits, so it does seem by design. We should open a discussion about this

File: boards/arm/volteer/Kconfig.board [hunk 11]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_VOLTEER
```
Comment (jettr) [inline boards/arm/volteer/Kconfig.board:4]: You are your thoughts about working on the downstream version of volteer for now until we get things working. Then we will definitely want to upstream this board.

File: include/dt-bindings/gpio/gpio.h (hunk 10 above)
Comment (pabigot) [reply include/dt-bindings/gpio/gpio.h:10]: This is indeed by design: Zephyr only allows devicetree use of the specifier flags that are supported in devicetree files by Linux.  I believe the intent is to retain devicetree source file compatibility independent of OS.  Best to open an RFC issue if you want that changed.

**Comment by andrewboie** [comment]
can you update the PR description and title with a concise summary of what this pull request is doing?

File: boards/arm/volteer/Kconfig.board (hunk 11 above)
Comment (vphirema) [reply boards/arm/volteer/Kconfig.board:4]: ACK, I will delete this CL

File: include/dt-bindings/gpio/gpio.h (hunk 10 above)
Comment (vphirema) [reply include/dt-bindings/gpio/gpio.h:10]: ACK, i will file a RFC

File: include/power/ndsx_espi.h (hunk 7 above)
Comment (vphirema) [reply include/power/ndsx_espi.h:18]: ACK, will change it

File: include/power/x86_non_dsx.h (hunk 8 above)
Comment (vphirema) [reply include/power/x86_non_dsx.h:109]: ACK, will change it

File: include/dt-bindings/gpio/gpio.h (hunk 10 above)
Comment (vphirema) [reply include/dt-bindings/gpio/gpio.h:10]: submitted an RFC, I don't have access to add people 
https://github.com/zephyrproject-rtos/zephyr/issues/29908

**Comment by vphirema** [comment]
> can you update the PR description and title with a concise summary of what this pull request is doing?
I have added a feature Feature request 
https://github.com/zephyrproject-rtos/zephyr/issues/29909

**Comment by andrewboie** [comment]
> can you update the PR description and title with a concise summary of what this pull request is doing?
> I have added a feature Feature request
> #29909

No, I'm serious. What you have right now still reads:

> No description provided.

When submitting PRs please provide a paragraph or two summarizing what this commit changes, why the changes were made, and links to any supporting GH issues.

The title of the PR should also be descriptive and not just be whatever the (often heavily abbreviated) branch name is.

You are able to edit both the description and title of PRs even after they are submitted. Please do this. Sending PRs with empty description is unacceptable and is burdensome to reviewers.

**Comment by vphirema** [comment]
> > can you update the PR description and title with a concise summary of what this pull request is doing?
> > I have added a feature Feature request
> > #29909
> 
> No, I'm serious. What you have right now still reads:
> 
> > No description provided.
> 
> When submitting PRs please provide a paragraph or two summarizing what this commit changes, why the changes were made, and links to any supporting GH issues.
> 
> The title of the PR should also be descriptive and not just be whatever the (often heavily abbreviated) branch name is.
> 
> You are able to edit both the description and title of PRs even after they are submitted. Please do this. Sending PRs with empty description is unacceptable and is burdensome to reviewers.

ACK. I haven't modified the code/commit yet. Do I need to wait until the Feature request gets some attention or can submit the code before that?

**Pull request closed without merging**